src/cast_core/src/CommitLatency.cpp

```diff
@@ -0,0 +1,195 @@
+
+#include <memory>
+
+#include <yaml-cpp/yaml.h>
+
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/database.hpp>
+
+#include <boost/log/trivial.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/ExecutionStrategy.hpp>
+#include <gennylib/InvalidConfigurationException.hpp>
+#include <gennylib/context.hpp>
+#include <value_generators/value_generators.hpp>
+
+
+namespace genny::actor {
+
+struct CommitLatency::PhaseConfig {
+    mongocxx::collection collection;
+    mongocxx::write_concern wc;
+    mongocxx::read_concern rc;
+    mongocxx::read_preference rp;
+    std::string rp_string;
+    std::shared_ptr<mongocxx::client_session> session;
+    mongocxx::options::find optionsFind;
+    mongocxx::options::update optionsUpdate;
+    mongocxx::options::aggregate optionsAggregate;
+    bool useSession;
+    int64_t repeat;
+    int64_t threads;
+    std::uniform_int_distribution<int64_t> amountDistribution;
+    ExecutionStrategy::RunOptions options;
+
+    PhaseConfig(PhaseContext& phaseContext, const mongocxx::database& db)
+        : collection{db[phaseContext.get<std::string>("Collection")]},
+          rp_string{phaseContext.get<std::string>("ReadPreference")},
+          useSession{phaseContext.get<bool, false>("Session")},
+          repeat{phaseContext.get<IntegerSpec>("Repeat")},
+          threads{phaseContext.get<IntegerSpec>("Threads")},
+          amountDistribution{-100, 100},
+          options{ExecutionStrategy::getOptionsFrom(phaseContext, "ExecutionsStrategy")} {
+              // write_concern
+              if( phaseContext.get<bool, false>("WriteConcernMajority") ) {
+                  wc.majority(std::chrono::milliseconds{0});
+              }
+              else {
+                  wc.nodes( (std::int32_t) phaseContext.get<IntegerSpec, false>("WriteConcern").value_or(1) );
+              }
+              if( phaseContext.get<bool, false>("WriteConcernJournal") ) {
+                  wc.journal(true);
+              }
+              if( ! phaseContext.get<bool, false>("WriteConcernJournal").value_or(true) ) {
+                  // On some mongod versions it makes a difference whether journal is unset vs explicitly set to false
+                  wc.journal(false);
+              }
+              optionsUpdate.write_concern(wc);
+
+              // read_concern
+              rc.acknowledge_string(phaseContext.get<std::string>("ReadConcern"));
+              // Ugh... read_concern cannot be set for operations individually, only through client, database, or collection.
+              // And out of those, only client is valid inside transactions. (DRIVERS-619)
+              // It seems mongocxx::database doesn't provide a getter to get the client, nor is the client available in this context.
+              // Ergo, transactions cannot be supported...
+              // TODO: Should set _client. (And options::find etc should support read_concern!)
```

Comment:
Similar to how you pass in the `mongocxx::database` you could pass in the `mongocxx::client` as well?

Reply:
Cool. Will try that. I wasn't sure I can / allowed to change this interface.